packages/wrangler/src/__tests__/helpers/mock-upload-worker.ts

```diff
@@ -57,7 +58,10 @@ export function mockUploadWorkerRequest(
 		const metadata = JSON.parse(
 			await toString(formBody.get("metadata"))
 		) as WorkerMetadata;
-		if (expectedType === "esm") {
+		if (expectedExperimentalAssetsForm) {
```

Comment:
Yeah, I think this is also a bit of a symptom of my earlier comment. To me, it doesn't feel like this belongs to this if/else. The if/else here validates the two possible ways of specifying an entrypoint to the Worker. Independently of that, we may or may not have presence of an `assets` key in the metadata that we want to validate.

Reply:
added a third 'none' entry type for the static assets only case